File: src/main/java/org/example/command/CommandHandler.java
package org.example.command;

import org.example.command.api.Commands;
import org.example.query.domain.DomainModel;

public class CommandHandler implements Commands {

    DomainModel domainModel;

    public CommandHandler(DomainModel domainModel) {
        this.domainModel = domainModel;
    }

    @Override
    public void createItem(String id) {
        domainModel.createItem(id);
    }

    @Override
    public void createItem(String id, int[] position, int value) {
        domainModel.createItem(id, position, value);
    }

    @Override
    public void deleteItem(String id) {
        domainModel.deleteItem(id);
    }

    @Override
    public void moveItem(String id, int[] vector) {
        domainModel.moveItem(id, vector);
    }

    @Override
    public void changeValue(String id, int newValue) {
        domainModel.changeValue(id, newValue);
    }
}

File: src/main/kotlin/command/impl/CommandHandler.kt
package command.impl

import command.api.Commands
import command.api.DomainModel

class CommandHandler(private val domainModel: DomainModel): Commands {

    override fun createItem(id: String) {
        domainModel.createItem(id)
    }

    override fun createItem(id: String, position: IntArray, value: Int) {
        domainModel.createItem(id, position, value)
    }

    override fun deleteItem(id: String) {
        domainModel.deleteItem(id)
    }

    override fun moveItem(id: String, vector: IntArray) {
        domainModel.moveItem(id, vector)
    }

    override fun changeValue(id: String, newValue: Int) {
        domainModel.changeValue(id, newValue)
    }
}
